src/shared/Core/CommandContext.cs

```diff
@@ -22,6 +23,11 @@ public interface ICommandContext : IDisposable
         /// </summary>
         string InstallationDirectory { get; }
 
+        /// <summary>
+        /// Application execution timer.
+        /// </summary>
+        Stopwatch ApplicationTimer { get; }
```

Comment:
If this global timer is being added for Trace2 reasons, I'd add Trace2 to the name of the variable to make it more obvious.

Reply:
Feels like `Trace2` should own the timer. In this case, `Stopwatch` should either be created inside of `Main` and passed to the `CommandContext.ctor` (which passes it through to `Trace2.ctor`), or `Trace2` should create its own `Stopwatch`, and `Trace2` should be created directly inside `Main` and passed to `CommandContext.ctor`.

Reply:
Or.. why not just store the `DateTime` or `long` Unix timestamp of when the application started, and compute elapsed times as we process events?

`DateTime.Now - DateTime.UnixEpoch`

or

`DateTimeOffset.Now.ToUnixTimeSeconds()`

Reply:
BTW.. does Git use local time, or UTC?

Reply:
UTC

Reply:
Each event record needs a "time" field with the current time (UTC with hopefully microsecond resolution).

Some event records add a "t_abs" field of elapsed seconds (float with 6 digits behind the decimal point).
(Yes, this is redundant, but it was helpful to have both in the format.)

Either way works, passing `DateTime.NowUtC()` (or whatever the c# class/method is called) on
the `Write<event>()` call is more direct and has less latency.  Letting the trace helper thread do the
call to NowUTC() is easier, but less accurate since a small amount of time has elapsed because of the
message queuing and background thread handoff.  (But it's not that bad either way.)

Maybe have a "trace2 init / ctor" call in/near Main() that sets a private start-time in the class (and sets
the SID) -- and maybe hides both from the rest of the program.